tests/test_magic.py

```diff
@@ -19,6 +19,7 @@
 from time import sleep
 from threading import Thread
 from unittest import TestCase, mock
+import nbformat
```

Comment:
This needs to be imported after `pytest.importorskip("nbformat")` as in:

https://github.com/ipython/ipython/blob/6cc4d06d0154b12a9d6fa36dad1a14920480c423/tests/test_run.py#L396-L399